backend/openapi.yaml

```diff
@@ -2345,18 +2345,28 @@ paths:
   /api/v1/thirty-year-regulation:
 
     get:
-      description: Check housing companies over 30 years old whether they should be released from hitas regulation or not.
-      operationId: check-regulation
+      description: >-
+        If 'check' parameter is 'true', check housing companies over 30 years old
+        whether they should be released from Hitas regulation or not.
+        Otherwise, try to fetch existing regulation results.
+      operationId: read-or-check-regulation
```

Comment:
So "check" is actually not checking _(tarkastaa)_, but running the regulation? Maybe some rewording could be in order..

Also it would be better to split running the regulation in the same endpoint from fetching the results. a GET endpoint is generally never used to modify data. I would the "run regulation" apart to a GET and POST endpoint

Reply:
Yes, it would make more sense to split those into two endpoints, I'll implement that.